apps/eth_blockchain/lib/eth_blockchain/transaction.ex

```diff
@@ -0,0 +1,170 @@
+# Copyright 2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule EthBlockchain.Transaction do
+  @moduledoc false
+
+  import Utils.Helpers.Encoding
+
+  alias Keychain.Signature
+  alias ExthCrypto.Hash.Keccak
+  alias EthBlockchain.{Adapter, ABI}
+
+  defstruct nonce: 0,
+            gas_price: 0,
+            gas_limit: 0,
+            to: <<>>,
+            value: 0,
+            v: nil,
+            r: nil,
+            s: nil,
+            init: <<>>,
+            data: <<>>
+
+  @spec send_eth(tuple(), atom() | nil, pid() | nil) ::
+          {atom(), String.t()} | {atom(), atom()} | {atom(), atom(), String.t()}
+  def send_eth(params, adapter \\ nil, pid \\ nil)
+
+  def send_eth({from_address, to_address, amount}, adapter, pid) do
+    gas_price = Application.get_env(:eth_blockchain, :default_gas_price)
+    send_eth({from_address, to_address, amount, gas_price}, adapter, pid)
+  end
+
+  def send_eth({from_address, to_address, amount, gas_price}, adapter, pid)
+      when byte_size(from_address) == 42 and byte_size(to_address) == 42 and is_integer(amount) and
+             is_integer(gas_price) do
+    case get_transaction_count({from_address}, adapter, pid) do
```

Comment:
Does `from_address` need to be wrapped in a tuple?

Reply:
This is just to keep it consistent with other similar calls.